tools/generate_tables_sql_and_testNames.php

```diff
@@ -1,118 +1,234 @@
 #!/usr/bin/php
 <?php
-
 /**
- * generate_tables_sql.php takes the ip_output.txt file generated from
- * quickform_parser.php and outputs an sql build file for the table of each
- * instrument it finds in the ip_output.txt file.  These sql files are output
- * to the tables_sql/ subdirectory.
+ * Behavioural
+ *
+ * PHP Version 7
  *
- * ex cmd:  php generate_tables_sql.php
+ * Parses a LINST file and creates a corresponding SQL script. Optionally 
+ * applies this new script automatically.
  *
- * @package behavioural
+ * @category Tools
+ * @package  Behavioural
+ * @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
  */
-
-//Ensure php version compatability
-//taken from php.net notes
-
-// require all relevant OO class libraries
-require_once __DIR__ . "/../vendor/autoload.php";
-require_once "../php/libraries/Database.class.inc";
-require_once "../php/libraries/NDB_Config.class.inc";
-require_once "../php/libraries/NDB_BVL_Instrument.class.inc";
-
-$data = stream_get_contents(STDIN);
-
-$instruments=explode("{-@-}",trim($data));
-
-$tblCount=0;
-$parameterCount=0;
-$pages = array();
-foreach($instruments AS $instrument){
-    $catId="";
-    $items=explode("\n",trim($instrument));
-    foreach($items AS $item){
-        $paramId="";
-        $bits=explode("{@}",trim($item));
-        if(preg_match("/Examiner[0-9]*/" , $bits[1])){
-            continue;
-        }
-        switch($bits[0]){
-            //generate the CREATE TABLE syntax
+require_once 'generic_includes.php';
+// Display usage message if flag parameter is absent or invalid.
+if (!isset($argv[2]) 
+    || !in_array(
+        $argv[2],
+        array(
+            '--to-sql',
+            '--apply'
+        ),
+        true
+    )
+) {
+    $usage = <<<USAGE
+The generate_tables_sql_and_testNames.php script takes the 
+{instrument_name}.linst file from the /instrument_manager  module and installs 
+an instrument for the table of each instrument it finds in the 
+{instrument_name}.linst file.
+These sql files are written to the tables_sql/ subdirectory.
+
+Ex cmd without SQL execution:
+php generate_tables_sql_and_testNames.php
+../instruments/[instrument_name].linst --to-sql
+
+Ex cmd with SQL execution:
+php generate_tables_sql_and_testNames.php 
+../instruments/[instrument_name].linst --apply
+USAGE; 
+
+    echo $usage;
+};
+
+$execute_mysql_query = $argv[2] === '--apply';
+$data          = file_get_contents($argv[1]);
+$instruments   = explode("{-@-}", trim($data));
+$error_message = "";
+// Do validation
+if (!is_array($instruments)) {
+    $error_message = "Instrument file could not be processed.";
+} else {
+    $pages = array();
+    $sql_query_statement  = "";
+    $sql_query_table_name = "";
+    $sql_query_columns    = array();
+    // Setup Database object.
+    $factory   = \NDB_Factory::singleton();
+    $db        = $factory->database();
```

Comment:
generic includes should already give you a database

Reply:
I used code I previously saw in LORIS when writing this and it was awhile ago. It should be updated.

Reply:
yeah.. im trying to clean these up, you can take a look at some of my tool refactoring PRs to get an idea